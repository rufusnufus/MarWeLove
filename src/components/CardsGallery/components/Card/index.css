.card {
  width: 250px;
  height: 300px;
  background-color: white;
  overflow: hidden;
  border-radius: 10px;
  cursor: pointer;
}

.card:hover {
  scale: 1.1;
}

.card__image {
  height: 250px;
  background-size: cover;
}

.card__caption {
  line-height: 50px;
  text-align: center;
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
}
