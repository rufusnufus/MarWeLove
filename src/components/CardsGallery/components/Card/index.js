import './index.css';

function Card({ data }) {
  return (
    <div className="card">
      <div className="card__image" style={{'background-image': `url(${data.imageUrl})`}}></div>
      <div className="card__caption">
        {data.caption}
      </div>
    </div>
  );
}

export default Card;