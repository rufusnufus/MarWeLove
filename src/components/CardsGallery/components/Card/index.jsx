import React from 'react'
import { useHistory } from 'react-router-dom'
import Bookmark from '../../../Bookmark'

import './index.css'

function Card({ data, type }) {
  const history = useHistory()
  const handleClick = () => {
    history.replace(`/${type}/${data.id}`)
  }

  return (
    <div role="navigation" className="card" onClick={handleClick}>
      <Bookmark id={data.id} bookmark={data.bookmark} type={type} />
      <div
        className="card__image"
        data-testid="card__image"
        style={{ backgroundImage: `url(${data.thumbnail?.path}.${data.thumbnail?.extension})` }}
      />
      <div className="card__caption">{type === 'characters' ? data.name : data.title}</div>
    </div>
  )
}

export default Card
