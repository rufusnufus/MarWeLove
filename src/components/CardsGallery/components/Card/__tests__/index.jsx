/**
 * @jest-environment jsdom
 */

import React from 'react'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import '@testing-library/jest-dom/extend-expect'
import { cleanup, fireEvent, render } from '@testing-library/react'
import Card from '../index'
import mockStore from '../../../../../__mocks__/store'

let store
beforeEach(() => {
  store = mockStore()
})
afterEach(cleanup)

const mockData = {
  id: 1009148,
  thumbnail: {
    path: 'abc',
    extension: 'def'
  },
  name: 'character',
  title: 'comic'
}

const mockDataWithoutThumbnail = {
  id: 1009148,
  name: 'character',
  title: 'comic'
}

const mockHistoryReplace = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    replace: mockHistoryReplace
  })
}))

jest.mock('../../../../Bookmark', () => () => 'Bookmark')

describe('Card', () => {
  describe('Character', () => {
    it('Render', () => {
      const { queryByText, getByTestId } = render(
        <Provider store={store}>
          <Card type="characters" data={mockData} />
        </Provider>
      )

      expect(queryByText('character')).toBeTruthy()
      expect(getByTestId('card__image')).toHaveStyle(
        `background-image: url(${mockData.thumbnail.path}.${mockData.thumbnail.extension})`
      )
      expect(queryByText('character')).toMatchSnapshot()
    })

    it('Render without thumbnail', () => {
      const { queryByText, getByTestId } = render(
        <Provider store={store}>
          <Card type="characters" data={mockDataWithoutThumbnail} />
        </Provider>
      )

      expect(queryByText('character')).toBeTruthy()
      expect(getByTestId('card__image')).toHaveStyle(
        `background-image: url(${mockDataWithoutThumbnail.thumbnail?.path}.${mockDataWithoutThumbnail.thumbnail?.path})`
      )
      expect(queryByText('character')).toMatchSnapshot()
    })

    it('After click route path changes', () => {
      const { queryByText } = render(
        <Provider store={store}>
          <MemoryRouter>
            <Card type="characters" data={mockData} />
          </MemoryRouter>
        </Provider>
      )

      fireEvent.click(queryByText('character'))
      expect(mockHistoryReplace).toHaveBeenCalledWith('/characters/1009148')
    })
  })

  describe('Comic', () => {
    it('Render', () => {
      const { queryByText, getByTestId } = render(
        <Provider store={store}>
          <Card type="comics" data={mockData} />
        </Provider>
      )

      expect(queryByText('comic')).toBeTruthy()
      expect(getByTestId('card__image')).toHaveStyle(
        `background-image: url(${mockData.thumbnail.path}.${mockData.thumbnail.extension})`
      )
      expect(queryByText('comic')).toMatchSnapshot()
    })

    it('Render without thumbnail', () => {
      const { queryByText, getByTestId } = render(
        <Provider store={store}>
          <Card type="comics" data={mockDataWithoutThumbnail} />
        </Provider>
      )

      expect(queryByText('comic')).toBeTruthy()
      expect(getByTestId('card__image')).toHaveStyle(
        `background-image: url(${mockDataWithoutThumbnail.thumbnail?.path}.${mockDataWithoutThumbnail.thumbnail?.path})`
      )
      expect(queryByText('comic')).toMatchSnapshot()
    })

    it('After click route path changes', () => {
      const { queryByText } = render(
        <Provider store={store}>
          <MemoryRouter>
            <Card type="comics" data={mockData} />
          </MemoryRouter>
        </Provider>
      )

      fireEvent.click(queryByText('comic'))
      expect(mockHistoryReplace).toHaveBeenCalledWith('/comics/1009148')
    })
  })
})
