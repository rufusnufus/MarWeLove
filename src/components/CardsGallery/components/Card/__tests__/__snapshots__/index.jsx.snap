// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Character Render 1`] = `
<div
  class="card__caption"
>
  character
</div>
`;

exports[`Card Character Render without thumbnail 1`] = `
<div
  class="card__caption"
>
  character
</div>
`;

exports[`Card Comic Render 1`] = `
<div
  class="card__caption"
>
  comic
</div>
`;

exports[`Card Comic Render without thumbnail 1`] = `
<div
  class="card__caption"
>
  comic
</div>
`;
